include/envoy/http/async_client.h

```diff
@@ -62,6 +128,16 @@ class AsyncClient {
    */
   virtual Request* send(MessagePtr&& request, Callbacks& callbacks,
                         const Optional<std::chrono::milliseconds>& timeout) PURE;
+
+  /**
+   * Start an HTTP stream asynchronously
+   * @param callbacks the callbacks to be notified of stream status.
+   * @return a stream handle or nullptr if no stream could be started. NOTE: In this case
+   *         onResetStream() has already been called inline. The client owns the stream and
+   *         the handle can be used to send more messages or close the stream.
+   */
+  virtual Stream* start(StreamCallbacks& callbacks,
+                        const Optional<std::chrono::milliseconds>& timeout) PURE;
```

Comment:
It's questionable whether timeout makes sense on this interface, since the only reason I could see it really ever being used is for long lived streaming API requests like you are implementing it for. But I don't have a strong opinion and could go either way. I guess since it's optional, we can leave it. (Please update the comments though).